genesis/engine/solvers/rigid/constraint_noslip.py

```diff
@@ -36,10 +36,12 @@ def kernel_build_efc_AR_b(
             rigid_solver.func_solve_mass_batched(
                 constraint_state.Mgrad,
                 constraint_state.Mgrad,
+                constraint_state.Mgrad,  # this will not be used anyway because is_backward is False
```

Comment:
I don't get why specifying a different out variable for the value and its gradient is an option in the first place. It seems that it is never the case in your PR. If so, I would suggest removing this option that is adding complexity (and therefore increasing maintenance burden because in theory it should be unit-tested) without any benefit.